compiler-core/src/io.rs

```diff
@@ -62,6 +62,62 @@ impl Utf8Writer for String {
     }
 }
 
+/// A wrapper for any writer that keeps track of what line/column it is
+/// currently on.
+///
+/// Used to produce SourceMaps.
+#[derive(Debug)]
+pub struct CursorPositionWriter<'a, W> {
```

Comment:
Or move this to `compiler-core/src/io/cursor_position_writer.rs` ?
Also not a fan of the name